satpy/readers/pmw_channels_definitions.py

```diff
@@ -0,0 +1,415 @@
+                     self.central + self.side + self.bandwidth/2. >=
+                     other.central + other.side + other.bandwidth/2.))
+
+        return False
+
+    def distance(self, value):
+        """Get the distance from value."""
+        if self == value:
+            try:
+                left_side_dist = abs(value.central - value.side - (self.central - self.side))
+                right_side_dist = abs(value.central + value.side - (self.central + self.side))
+                return min(left_side_dist, right_side_dist)
+            except AttributeError:
+                if isinstance(value, (tuple, list)):
+                    return abs((value[0] - value[1]) - (self.central - self.side))
+
+                left_side_dist = abs(value - (self.central - self.side))
+                right_side_dist = abs(value - (self.central + self.side))
+                return min(left_side_dist, right_side_dist)
+        else:
+            return np.inf
+
+    @classmethod
+    def convert(cls, frq):
+        """Convert `frq` to this type if possible."""
+        if isinstance(frq, dict):
+            return cls(**frq)
+        return frq
+
+
+class FrequencyRangeBase(NamedTuple):
+    """Base class for frequency ranges.
+
+    This is needed because of this bug: https://bugs.python.org/issue41629
+    """
+
+    central: float
+    bandwidth: float
+    unit: str = "GHz"
+
+
+class FrequencyRange(FrequencyRangeBase):
+    """The Frequency range class.
+
+    The elements of the range are central and bandwidth values, and optionally
+    a unit (defaults to GHz). No clever unit conversion is done here, it's just
+    used for checking that two ranges are comparable.
+
+    This type is used for passive microwave sensors.
+
+    """
+
+    def __eq__(self, other):
+        """Return if two channel frequencies are equal.
+
+        Args:
+            other (tuple or scalar): (central frq, band width frq) or scalar frq
+
+        Return:
+            True if other is a scalar and min <= other <= max, or if other is
+            a tuple equal to self, False otherwise.
+
+        """
+        if other is None:
+            return False
+        if isinstance(other, numbers.Number):
+            return other in self
+        if isinstance(other, (tuple, list)) and len(other) == 2:
+            return self[:2] == other
+        return super().__eq__(other)
+
+    def __ne__(self, other):
+        """Return the opposite of `__eq__`."""
+        return not self == other
+
+    def __lt__(self, other):
+        """Compare to another frequency."""
+        if other is None:
+            return False
+        return super().__lt__(other)
+
+    def __gt__(self, other):
+        """Compare to another frequency."""
+        if other is None:
+            return True
+        return super().__gt__(other)
+
+    def __hash__(self):
+        """Hash this tuple."""
+        return tuple.__hash__(self)
```

Comment:
Seems to repeat lines 94–108.  Could this be refactored out?

Reply:
That is correct. I also discussed with @mraspaud that one could probably define the quadruple side band from the double side band, and then use inheritance to get rid of this redundancy. However, I am not sure we will ever find use with all the arithmetics on these bands. For instance I have chosen not to try to be too smart when calculating distances etc between them. Thus I would rather leave such refactoring till we really see a need for this.
I also don't have much time/resources for now to elaborate much further on this.

Reply:
Good point! I did this just now.